components/raftstore/src/store/peer.rs

```diff
@@ -3830,6 +3838,9 @@ where
             // The message is dropped silently, this usually due to leader absence
             // or transferring leader. Both cases can be considered as NotLeader error.
             return Err(Error::NotLeader(self.region_id, None));
+        } else if self.force_leader {
+            // forward the commit index
+            self.raft_group.raft.raft_log.committed = self.raft_group.raft.raft_log.last_index();
```

Comment:
I suggest to return error instead of committing any logs.

Reply:
we need to commit the new proposal, e.g. remove the failed peer, in force leader state